### PR TITLE
Aerogear 10238

### PR DESCRIPTION
**Pull request by Aindriunannerb**
## Motivation
https://issues.redhat.com/browse/AEROGEAR-10238
## What
Implement a delete button that when clicked, it would open up a dialog that requires the user to input the name of the app to deleted for confirmation.
when the delete app button is clicked the app is deleted, the dialog is closed and the application list is refreshed. 
 
Comparison view the dialogs

Old delete app view 
old Delete

New Delete app view 
Delete app small

Butons right aligned and showing hover
buttons showing hover

### CODE REVIEW

File: src/application/wizard/DeleteApplicationPage.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      console.log(err);
+    }
+  };
+
+  render(): React.ReactNode {
+    return (
+      <EmptyState>
+        <Form className="dialog-form">
+          <FormGroup
+            label="Delete Application"
```
Comment (ziccardi) [inline src/application/wizard/DeleteApplicationPage.tsx:52]: This should be the title of the Dialog

File: src/application/wizard/DeleteApplicationPage.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            fieldId="simple-form-title"
+            helperText={`Do you really want to delete "${
+              this.props.app!.name
```
Comment (ziccardi) [inline src/application/wizard/DeleteApplicationPage.tsx:55]: Can you make this bold? (only the app name)

File: src/common/Modal.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  render(): React.ReactNode {
+    return (
+      <Modal
+        isLarge
+        hideTitle={true}
+        title="is hidden"
+        isOpen={this.props.open}
+        onClose={this.props.close}
+      >
+        {this.props.children}
```
Comment (ziccardi) [inline src/common/Modal.tsx:19]: This component doesn't give much added value compared to the `Modal` component itself.
I believe we can remove this and put it inside the `DeleteApplicationPage` component directly

File: src/landing/ApplicationList.tsx [hunk 4]
```diff
@@ -90,6 +118,20 @@ export class ApplicationList extends Component<Props, State> {
...
+                }}
+              >
+                <DeleteApplicationPage
+                  app={this.state.selectedApp}
+                  close={() => {
+                    this.setState({ dialogModal: false });
+                    refresh();
+                  }}
+                />
+              </DialogModal>
```
Comment (ziccardi) [inline src/landing/ApplicationList.tsx:134]: See previous comment

**Comment by ziccardi** [comment]
@Aindriunannerb in the screenshot, I see that the dialog is very large, much larger than its content. We can maybe make it a bit smaller and/or make the content fill the entire width.
Moreover the `Delete Application` title, should be the Dialog title, so that it is aligned with the `X` button that closes the dialog.

**Comment by ziccardi** [comment]
@Aindriunannerb 
The old implementation of the list was like this
old impl

The new one is like this
new impl

Can we make the new one look similar to the old one?
Alternatively, can we horizontally align the button on the right and center them vertically?

To avoid polluting the page would be nice to have them appear only when overing over the application as in the previous implementation

**Comment by ziccardi** [comment]
@Aindriunannerb Both EDIT and DELETE button opens the DELETE dialog. Is that expected?

**Comment by Aindriunannerb** [comment]
> @Aindriunannerb Both EDIT and DELETE button opens the DELETE dialog. Is that expected?

this is resolved in another PR when the Edit functionality is implemented correctly.

**Pull request merged**